fonts

Auhseh opened the pull request:
<!--
## Submitting a Pull Request
We love contributions and appreciate any help you can offer!
-->

## Related Issue

<!-- Include a link to the issue (e.g. Resolves #12) -->

## Summary of Changes

<!-- Briefly summarize the changes made, lists are appreciated, ideally with checklists

1. [x] Thing I fixed
1. [x] Other thing I updated
1. [x] Docs I updated
-->

thescientist13 commented on src/stories/documentation.mdx:
```diff
@@ -0,0 +1,11 @@
+## GreenwoodJS Website Documentation
+
+
+We use a combination of technologies to build the GreenwoodJS website
```
Looks like the formatting is a bit off here and there aren't any links, so let's make sure to get those in
![Screenshot 2024-04-13 at 6 19 26 PM](https://github.com/ProjectEvergreen/www.greenwoodjs.dev/assets/895923/e324624d-08aa-4b90-9c31-275e5c564a6e)

You can check the source from the linked reference for an example
https://raw.githubusercontent.com/AnalogStudiosRI/www.blissfestri.com/main/src/stories/Documentation.stories.mdx

Also, I think if we can have these side links uppercased (I think we just need to rename the files to _Documentation_ etc, that would be ideal.

thescientist13 commented on src/stories/styleguide.mdx:
```diff
@@ -0,0 +1,70 @@
+
+
+## Colors
+The following theme colors available are:
```
We'll definitely want to flesh this section out with examples like in the reference storybook
https://raw.githubusercontent.com/AnalogStudiosRI/www.blissfestri.com/main/src/stories/Styleguide.stories.mdx

thescientist13 commented on src/stories/styleguide.mdx:
```diff
@@ -0,0 +1,70 @@
+## Typography
+The whole website uses Vercel's **Geist** font in the following sizes:
+
+`--hero-big`
```
For the design system tokens, I don't think the goal is to try and capture every token used throughout all pages like these `--hero` specific ones, but rather just the ones all other tokens should build / compose off of.

I think just the main colors, fonts, sizes, etc should be good to start out with.  We don't need to predict every token right now, and this can totally evolve and grow as we build out the rest of the pages.  🏃‍♂️

thescientist13 commented on src/stories/styleguide.mdx:
```diff
@@ -0,0 +1,70 @@
+
+
+## Spacing
+The whole website is built on an 8pt Grid, and spacing is always in multiples of 8px
```
If we have any tokens for this, it would be good to document them here.

thescientist13 commented on src/styles/theme.css:
```diff
@@ -1,8 +1,38 @@
+  --color-white: #FFFFFF;
+  --font-primary: 'Geist', 'sans-serif';
+  --font-size-base: 16px;
+  --container-radius: 6rem;
```
similar to my comments above, we don't have to capture _page / component_ specific tokens for the design system, we can start with just the ones that apply to broadest parts of the design that should be used everywhere, like colors and fonts.  This can always grow as the site grows.

thescientist13 reviewed (commented):
This is a good start, thanks!

I took the opportunity to source the fonts from npm and can confirm it is working both for development and Storybook preview.  Formatted the PR description a bit.

I left some feedback to consider but let me know if you need any help on it, I can help with some of them if unclear:
1. [ ] Limit the design tokens / CSS variables to just the most common / global ones we'll need for now
1. [ ] Just need a little formatting and links for the Documentation page
1. [ ] For the Styleguide docs, we'll need to curate the design tokens list and then show examples (like you can see in the reference example)
1. [ ] For the tech stack docs, just take what you see from the reference link and just remove the Tailwind mention (and the part about EventBrite and Netlify Forms since they don't apply to this project) 

----

Will mark this PR as a draft as we get through the feedback items.

thescientist13 commented on src/styles/theme.css:
```diff
@@ -1,8 +1,29 @@
+  --color-accent: #C3DDBA;
+  --color-white: #FFFFFF;
+  --color-black: #020202;
+  --color-gray: #F5F5F5;
```
@Auhseh 
What is the distinction between the `--color-gray` and `--color-inner-gray`.  I'm guessing these are intended for the docs pages?

thescientist13 replied:
@Auhseh 
Any thoughts here?  I'm wondering if we just need better names for these, or are these just one-offs to a particular page?

thescientist13 commented on src/styles/theme.css:
```diff
@@ -1,8 +1,29 @@
+  --color-gray-inner: #EDEFEC;
+  --font-primary: 'Geist-Sans', system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;;
+  --font-secondary: 'Geist-Mono', monospace;
+  --font-size-base: 16px;
```
@aholtzman 
would be interested to know your thoughts on this baseline font setting and the variables we have outlined here, and any feedback you may have.  👍

thescientist13 commented:
Thanks @Auhseh , this is looking good!  I just made some updates to get the colors and fonts showing in Storybook

![Screenshot 2024-04-17 at 8 45 08 PM](https://github.com/ProjectEvergreen/www.greenwoodjs.dev/assets/895923/ff889d5f-4a3b-4148-82f4-d5eb8f6bde50)
![Screenshot 2024-04-17 at 8 49 41 PM](https://github.com/ProjectEvergreen/www.greenwoodjs.dev/assets/895923/d8ed679b-417e-440c-bb5d-571d59029549)

Left some comments in-line for you, other than this one: what is the _Grid_wh.svg_ for?

thescientist13 reviewed (commented):
@Auhseh 
It would be good if we get as many of the images we think we'll need in now if we're planning on committing a bunch of them.  I think at least for the header we would need
- Discord
- X / Twitter
- "App Launcher" (or whatever we call the icon we show when the header navigation is in mobile view)

Auhseh commented:
> @Auhseh It would be good if we get as many of the images we think we'll need in now if we're planning on committing a bunch of them. I think at least for the header we would need
> 
> * Discord
> * X / Twitter
> * "App Launcher" (or whatever we call the icon we show when the header navigation is in mobile view)

Added the other assets and also took the liberty to add a short ReadMe with a guide on how to use the assets in code.

Auhseh commented:
> Thanks @Auhseh for all this , looking good pretty good!
> 
> I just made some updates to get the colors and fonts showing in Storybook
> 
> ![Screenshot 2024-04-17 at 8 45 08 PM](https://private-user-images.githubusercontent.com/895923/323430377-ff889d5f-4a3b-4148-82f4-d5eb8f6bde50.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM1MTQzMjcsIm5iZiI6MTcxMzUxNDAyNywicGF0aCI6Ii84OTU5MjMvMzIzNDMwMzc3LWZmODg5ZDVmLTRhM2ItNDE0OC04MmY0LWQ1ZWI4ZjZiZGU1MC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQxOVQwODA3MDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iMzI5YTcyZmUzZWNkZWJhOTJlY2Q0Y2MwMTRiMWFmNWFjYWFhYzhkNmU4MThjMDY4MDNhNTU4ZGI1OWU5NzBiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.nWZPhI2OU3Jp3qHnhpPjKN7IfP9JJrTm6OqKEEjF7ek) ![Screenshot 2024-04-17 at 8 49 41 PM](https://private-user-images.githubusercontent.com/895923/323430379-d8ed679b-417e-440c-bb5d-571d59029549.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM1MTQzMjcsIm5iZiI6MTcxMzUxNDAyNywicGF0aCI6Ii84OTU5MjMvMzIzNDMwMzc5LWQ4ZWQ2NzliLTQxN2UtNDQwYy1iYjVkLTU3MWQ1OTAyOTU0OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQxOVQwODA3MDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03ZjMxYTkxNjMwOTlmNDM0NDJjMGU4ZmE4ZTRlZTMxNDlmMWFhMWM0OTA3NGZmZGFlMTMwMTg5NDI0ZTljMzNiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.K03GDU8jWmI4A8PAZANZ74roh9Hd_hAAnSWGZycppjU)
> 
> Left some comments in-line for and this one: what is the _Grid_wh.svg_ for?
> 
> (also, I renamed the assets to be consistent casing and using `-` for the space, just a bit easier to have all the files named the same way)

Thanks, it's the Grid background, Renamed it and added context in the ReadMe

Auhseh commented:
I think everything's pretty much ready, last thing would be to adding Description icons in the "Why Greenwood" section, pending when we articulate the copy.

thescientist13 commented:
Thanks, let me review.  Going forward, can we please keep the file names consistent?  I had updated them to be something like _some-name.png_.

I will fix the new ones added but just so we can keep the convention the same for everyone contributing to the project.  Thanks!

thescientist13 commented:
> Thanks, it's the Grid background, Renamed it and added context in the ReadMe

Ok, I was hoping maybe we could handle all those with CSS, but maybe we will need these as SVGs?

thescientist13 reviewed (approved):
This looks good!  Going to leave it open a couple more days for a final review, then will merge.  🚀

thescientist13 reviewed (commented):
As a side not, if anyone feels comfortable setting up build tools it might be a good time to start contributing on #24 as we start ramping up contributions.  Feel free to reach out in that issue if you have any questions about getting started.

thescientist13 commented on src/stories/Styleguide.mdx:
```diff
@@ -0,0 +1,149 @@
+
+## Spacing
+
+The whole website is built on an 8pt Grid, and spacing is always in multiples of 8px
```
Or maybe we can just delegate to Open Props
https://open-props.style/#sizes

thescientist13 replied:
Will delegate to this higher level conversation, this is a good start for getting things up and running for contributions - https://github.com/ProjectEvergreen/www.greenwoodjs.dev/discussions/7#discussioncomment-7645776

thescientist13 commented on src/styles/theme.css:
```diff
@@ -1,8 +1,29 @@
+
+* {
+  font-family: var(--font-primary);
+  font-size: var(--font-size-base);
```
Perhaps we should also add `box-sizing` too?
https://css-tricks.com/almanac/properties/b/box-sizing/

thescientist13 replied:
Added!

thescientist13 reviewed (approved):
This is great!  I'll plan on following up with [Linting / Formatting / Contribution Guidelines](https://github.com/ProjectEvergreen/www.greenwoodjs.dev/issues/24) next and then at least that should allow to make an initial start on our [Development Roadmap](https://github.com/ProjectEvergreen/www.greenwoodjs.dev/issues/14).

I'll also make sure to get a basic strategy documented around a more formal [CSS Strategy](https://github.com/ProjectEvergreen/www.greenwoodjs.dev/issues/7) as well.
